onpremise/onpremise.go

```diff
@@ -171,25 +178,102 @@ func WithCustomLogger(logger LogWriter) EngineOptions {
+	return func(cfg *Engine) error {
+		cfg.isFileWatcherEnabled = enabled
+		return nil
+	}
+}
+
+// ToggleUpdateOnStart enables or disables update on start
+// default is true
+// if enabled, engine will pull the data file from the distributor on start of the engine
+func ToggleUpdateOnStart(enabled bool) EngineOptions {
+	return func(cfg *Engine) error {
+		cfg.isUpdateOnStartEnabled = enabled
 
+		return nil
+	}
 }
 
-func New(config *dd.ConfigHash, opts ...EngineOptions) (*Engine, error) {
-	manager := dd.NewResourceManager()
+// ToggleAutoUpdate enables or disables auto update
+// default is true
+// if enabled, engine will automatically pull the data file from the distributor
+// if disabled, engine will not pull the data file from the distributor
+// options like WithDataUpdateUrl, SetLicenceKey will be ignored since auto update is disabled
+func ToggleAutoUpdate(enabled bool) EngineOptions {
+	return func(cfg *Engine) error {
+		cfg.isAutoUpdateEnabled = enabled
+
+		return nil
+	}
+}
+
+// ToggleCreateTempDataCopy enables or disables creating a temp copy of the data file
+// default is true
+// if enabled, engine will create a temp copy of the data file and use it to initialize the manager
+// if disabled, engine will use the original data file to initialize the manager
+// this is useful when 3rd party updates the data file on file system
+func ToggleCreateTempDataCopy(enabled bool) EngineOptions {
+	return func(cfg *Engine) error {
+		cfg.isCreateTempDataCopyEnabled = enabled
 
+		return nil
+	}
+}
+
+// SetTempDataDir sets the directory to store the temp data file
+// default is current directory
+func SetTempDataDir(dir string) EngineOptions {
+	return func(cfg *Engine) error {
+		dirFileInfo, err := os.Stat(dir)
+		if err != nil {
+			return fmt.Errorf("failed to get file path: %w", err)
+		}
+
+		if !dirFileInfo.IsDir() {
+			return fmt.Errorf("path is not a directory: %s", dir)
+		}
+
+		cfg.tempDataDir = dir
+		return nil
+	}
+}
+
+// Randomization sets the randomization time in seconds
+// default is 0
+// if set, when scheduling the file pulling, it will add randomization time to the interval
+// this is useful to avoid all engines pulling the data file at the same time in case of multiple engines/instances
+func Randomization(seconds int) EngineOptions {
+	return func(cfg *Engine) error {
+		cfg.randomization = seconds * 1000
+		return nil
+	}
+}
+
+func New(config *dd.ConfigHash, opts ...EngineOptions) (*Engine, error) {
 	pl := &Engine{
 		logger: logWrapper{
 			logger:  DefaultLogger,
 			enabled: true,
 		},
-		manager:     manager,
 		config:      config,
-		rdySignal:   make(chan error, 1),
 		stopCh:      make(chan struct{}, 1),
 		fileSynced:  false,
 		dataFileUrl: defaultDataFileUrl,
 		//default 15 minutes
-		dataFilePullEveryMs: 15 * 60 * 1000,
+		dataFilePullEveryMs:         30 * 60 * 1000,
+		isFileWatcherEnabled:        true,
+		isUpdateOnStartEnabled:      true,
```

Comment:
let's change this to false please, as reference implementation specifies it as false by default: https://github.com/51Degrees/pipeline-dotnet/blob/c75b438d0635a63e0fb562e7736659f61741ba73/FiftyOne.Pipeline.Engines/Constants.cs#L117